index.js

```diff
@@ -51,6 +51,23 @@ instance.prototype.incomingData = function(data) {
 		self.log('error', "incorrect username/password (expected no password)");
 		self.status(self.STATUS_ERROR, 'Incorrect user/pass');
 	}
+	// Heatbeat to keep connection alive (Working)
+	if (self.login === true && self.socket.connected) {
```

Comment:
Is it really needed to reinstall the heartbeat interval each time you receive data from the device? Whats the reason for this? Instead of doing it only when you are successfully logged in?

Reply:
No sorry, I did testing with with only self.login === true, but thought self.socket.connected would ensure there was a connection before sending command. My limited understanding of javascript.

So remove self.socket.connected?